localstack-core/localstack/services/sns/v2/models.py

```diff
@@ -143,6 +148,18 @@ def from_batch_entry(cls, entry: PublishBatchRequestEntry, is_fifo=False) -> "Sn
         )
 
 
+class PlatformEndpoint(TypedDict, total=False):
+    PlatformEndpointArn: str
+    Attributes: dict[str, str]
+    PlatformApplicationArn: str
```

Comment:
This type exists in the Api as well, but without the PlatformApplicationArn. Having a reference back to the owning object makes delete easier so we don't have to iterate over all applications, however it might not be worth losing the advantage of sticking exactly to the spec. Wdyt?

Reply:
Very good point! I believe @giograno is starting to establish a pattern here, where we use a `dataclass` to wrap the existing spec type, and add more data to it. 

See https://github.com/localstack/localstack-pro/pull/5397

Something like:

```python

from localstack.aws.api.sns import PlatformApplication

@dataclasses.dataclass
class PlatformApplicationDetails:
    platform_application: PlatformApplication
    platform_application_arn: str

```

I hope it's okay, I'll go ahead and push those to get the PR in this week :+1: